lib/config.js

```diff
@@ -229,6 +231,16 @@ function getLocalConfig(thisConfig, directory) {
             continue;
         }
 
+        // Check for root flag
+        if (localConfig.root === true) {
+            rootPath = path.dirname(localConfigFile);
+        }
+
+        // If root flag is set, don't consider file if it is above root
+        if (rootPath && !pathIsInside(path.dirname(localConfigFile), rootPath)) {
```

Comment:
That seems tricky tricky to me.  Does `.eslintrc` always come before `package.json`?  If so, then specifying root in `.eslintrc` would prevent configs in `package.json`, but not vice-versa.  That seems confusing and arbitrary to me.

Personally, I like the idea of specifying a root folder, and I don't think the implementation here is overly complex, is it?